staging/src/kubevirt.io/api/testdata/README.md

```diff
@@ -0,0 +1,155 @@
+
+```sh
+cd staging/src/kubevirt.io/api
+UPDATE_COMPATIBILITY_FIXTURE_DATA=true go test ./ -run //HEAD
+```
+
+### Previous versions
+
+The vX.Y.0 subdirectories contain serialized API objects from previous releases:
+
+```
+release-X.Y
+  <group>.<version>.<kind>.[json|yaml]
+```
+
+To run serialization tests for a previous version, like `v1.1.0`:
+
+```sh
+cd staging/src/kubevirt.io/api
+go test ./ -run //release-1.1
+```
+
+To run serialization tests for a particular group/version/kind, like `apps/v1` `Deployment`:
+```sh
+go test ./ -run /apps.v1.Deployment/
+```
+
+Example output:
+
+```    
+--- FAIL: TestCompatibility/kubevirt.io.v1.VirtualMachineInstance (0.01s)
+        --- FAIL: TestCompatibility/kubevirt.io.v1.VirtualMachineInstance/release-0.50 (0.01s)
+            compatibility.go:416: json differs
+            compatibility.go:417:   (
+                        """
+                        ... // 215 identical lines
+                                      "readonly": true
+                                    },
+                -                   "floppy": {
+                -                     "readonly": true,
+                -                     "tray": "trayValue"
+                -                   },
+                                    "cdrom": {
+                                      "bus": "busValue",
+                        ... // 678 identical lines
+                              "tscFrequency": -12
+                            },
+                -           "virtualMachineRevisionName": "virtualMachineRevisionNameValue"
+                +           "virtualMachineRevisionName": "virtualMachineRevisionNameValue",
+                +           "runtimeUser": 0
+                          }
+                        }
+                        """
+                  )
+                
+            compatibility.go:422: yaml differs
+            compatibility.go:423:   (
+                        """
+                        ... // 237 identical lines
+                                  pciAddress: pciAddressValue
+                                  readonly: true
+                -               floppy:
+                -                 readonly: true
+                -                 tray: trayValue
+                                io: ioValue
+                                lun:
+                        ... // 341 identical lines
+                          qosClass: qosClassValue
+                          reason: reasonValue
+                +         runtimeUser: 0
+                          topologyHints:
+                            tscFrequency: -12
+                        ... // 22 identical lines
+                        """
+                  )
+                
+```
+
+The above output shows that for VirtualMachineInstance:
+1. api-field: `spec.domain.devices.disks.floppy` was dropped. [ref-1](https://github.com/kubevirt/kubevirt/issues/2016)[ref-2](https://github.com/kubevirt/kubevirt/pull/2164)
```

Comment:
I am trying to understand how to interpret it to an action.

I am starting to understand that in order to interpret potential errors/warnings, one needs to understand what is allowed and what is not allowed to do when changing the API.
Do we have such a list? Can we come up with the parts that these tests are covering?